packages/url/src/test/index.test.js

```diff
@@ -30,28 +30,30 @@ import {
 } from '../';
 
 describe( 'isURL', () => {
-	it( 'returns true when given things that look like a URL', () => {
-		const urls = [
-			'http://wordpress.org',
-			'https://wordpress.org',
-			'HTTPS://WORDPRESS.ORG',
-			'https://wordpress.org/foo#bar',
-			'https://localhost/foo#bar',
-		];
-
-		expect( every( urls, isURL ) ).toBe( true );
-	} );
-
-	it( 'returns false when given things that don\'t look like a URL', () => {
-		const urls = [
-			'HTTP: HyperText Transfer Protocol',
-			'URLs begin with a http:// prefix',
-			'Go here: http://wordpress.org',
-			'http://',
-			'',
-		];
-
-		expect( every( urls, isURL ) ).toBe( false );
+	it.each( [
+		[ 'http://wordpress.org' ],
+		[ 'https://wordpress.org' ],
+		[ 'HTTPS://WORDPRESS.ORG' ],
+		[ 'https://wordpress.org/./foo' ],
+		[ 'https://wordpress.org/path?query#fragment' ],
+		[ 'https://localhost/foo#bar' ],
+		[ 'mailto:example@example.com' ],
+		[ 'ssh://user:password@127.0.0.1:8080' ],
+	] )( 'valid (true): %s', ( url ) => {
+		expect( isURL( url ) ).toBe( true );
+	} );
+
+	it.each( [
+		[ 'http://word press.org' ],
+		[ 'http://wordpress.org:port' ],
+		[ 'http://[wordpress.org]/' ],
+		[ 'HTTP: HyperText Transfer Protocol' ],
+		[ 'URLs begin with a http:// prefix' ],
+		[ 'Go here: http://wordpress.org' ],
+		[ 'http://' ],
+		[ '' ],
```

Comment:
> What about `https://www.wordpress#org` and `https://www.wordpress.#org`

We could include them, sure.

For context, I kept the existing tests, then added a few more based on the "Example" table in [Section 4 of the URL Living Standard](https://url.spec.whatwg.org/#urls), specifically adapting its "Failure" examples.

![URL parse results examples](https://user-images.githubusercontent.com/1779930/73093161-f3de8e00-3eab-11ea-8a57-7ef49216a3be.png)

It's interesting to consider there are many more examples there which are technically flagged as "not valid", but would still pass our test of `isURL`. It seems to be accepted that these would be subject to some internal corrections / normalizations.

That's not to say it contradicts what we're claiming to be considered as valid, because the specification itself mentions:

>A valid URL string defines what input would not trigger a validation error or failure when given to the URL parser.

Reply:
> What about `https://www.wordpress#org` and `https://www.wordpress.#org`

These are actually technically valid URLs.

In both cases, they behave as if `.wordpress` were a TLD (the latter, as a [fully qualified domain](https://en.wikipedia.org/wiki/Fully_qualified_domain_name)), with `#org` as the hash fragment.

It can be tested in your console:

```
> new URL( 'https://www.wordpress#org' );
URL {
  href: 'https://www.wordpress/#org',
  origin: 'https://www.wordpress',
  protocol: 'https:',
  username: '',
  password: '',
  host: 'www.wordpress',
  hostname: 'www.wordpress',
  port: '',
  pathname: '/',
  search: '',
  searchParams: URLSearchParams {},
  hash: '#org'
}
> new URL( 'https://www.wordpress.#org' );
URL {
  href: 'https://www.wordpress./#org',
  origin: 'https://www.wordpress.',
  protocol: 'https:',
  username: '',
  password: '',
  host: 'www.wordpress.',
  hostname: 'www.wordpress.',
  port: '',
  pathname: '/',
  search: '',
  searchParams: URLSearchParams {},
  hash: '#org'
}
```

Reply:
Oh yeh. Sorry. Was trying to think of things to break it. That chart looks like a good test case.